Add basic support for LG G4 (h815)

kamnxt opened the pull request:
Currently the screen doesn't work, but pmOS boots and can be accessed over SSH.

I've tried to disable KGSL, but it didn't seem to help...

MoreRobustThanYou commented:
try msm-fb-refresher

ollieparanoid commented on aports/device/linux-lg-h815/APKBUILD:
```diff
@@ -0,0 +1,99 @@
+# Kernel config based on: arch/arm64/configs/(CHANGEME!)
```
Please adjust this.

ollieparanoid commented on aports/device/linux-lg-h815/APKBUILD:
```diff
@@ -0,0 +1,99 @@
+	$pkgname-$_commit.tar.gz::https://github.com/LineageOS/${_repository}/archive/${_commit}.tar.gz
+	$_config
+	compiler-gcc6.h
+        fix-powerlevel.patch
```
Indentation: Please use tabs in APKBUILDs, we do this because that's how Alpine does it. We use spaces in our Python code.

ollieparanoid commented on aports/device/linux-lg-h815/APKBUILD:
```diff
@@ -0,0 +1,99 @@
+#		error "The following patches failed:"
+#		cat failed
+#		return 1
+#	fi
```
How about removing this commented out code?

kamnxt replied:
Oh whoops, was a bit quick when committing :P Will fix!

ollieparanoid commented on aports/device/linux-lg-h815/APKBUILD:
```diff
@@ -0,0 +1,99 @@
+pkgname="linux-lg-h815"
+pkgver=3.10.84
+pkgrel=0
+pkgdesc="LG LG G4 (h815) kernel fork"
```
One LG is enough :) I know this is auto-generated, it would be nice if you could change it.

kamnxt commented:
Fixed the issues you found and added the device repo (which I initially forgot to add). I've ran the sysfs dump script, and it seems like TWRP sets a bigger virtual_size for the framebuffer and uses panning, so perhaps that's related...

ollieparanoid commented:
I'm compiling the package now to confirm that it builds successfully, then I'll merge it.

Please create a [device page](https://wiki.postmarketos.org/wiki/Help:Device_Page).

kamnxt commented:
Created device page [here](https://wiki.postmarketos.org/wiki/LG_G4_International_(lg-h815))

ollieparanoid commented:
That's great, thanks! If you don't mind it would be even better if you could add it to the [overview](https://wiki.postmarketos.org/wiki/Devices), too.

kamnxt commented:
Done!

kamnxt commented:
Also, #789 can now be closed.

vrga commented:
for what its worth, i've managed to get it to run on my lg as well, can connect fine over ssh.
dmesg segfaults tho when i run it.

kamnxt commented:
Woo, nice to know someone tried it out already!
`dmesg` does indeed segfault for some reason. `logread` works though. Interestingly enough, `dmesg` didn't segfault before I got the splash screen and brightness working. I think I'll try to take a look at it again soon when I get time.

vrga commented:
What can i say, the idea of plasma mobile on this phone got me racing :D

Splash worked last night when i tried it but screen was turned off.
